src/icrar/leap-accelerate/algorithm/casa/PhaseRotate.cc

```diff
@@ -62,31 +64,61 @@ namespace icrar
 {
 namespace casalib
 {
-    void RemoteCalibration(MetaData& metadata, const std::vector<casacore::MVDirection>& directions)
+    // leap_remote_calibration
+    CalibrateResult Calibrate(
+        const casacore::MeasurementSet* ms,
+        MetaData& metadata,
+        const std::vector<casacore::MVDirection>& directions,
+        boost::optional<int> overrideStations,
+        int solutionInterval)
     {
-        auto input_queues = std::vector<std::queue<Integration>>(directions.size());
-        auto output_integrations = std::vector<std::queue<IntegrationResult>>(directions.size());
-        auto output_calibrations = std::vector<std::queue<CalibrationResult>>(directions.size());
+        auto output_integrations = std::make_unique<std::vector<std::queue<IntegrationResult>>>();
+        auto output_calibrations = std::make_unique<std::vector<std::queue<CalibrationResult>>>();
```

Comment:
Any particular reasons why these need to be pointers instead of plain vectors? ATM it will just add an extra allocation that seems unnecessary. This of course would also change the definition of `CalibrateResult`. There's no further usage of these returned values in this set of changes, but I'm assuming that in a real usage scenario the results will be used straight after `Calibrate` is called.

Reply:
I encountered issues when returning a pair of vectors which I assume to be related to return value optimization not happening on this function

Reply:
Would be worth double-checking. They way you are returning (e.g., `make_pair(std::move(v1), std::move(v2))`) should already prevent copying the vectors (they will be move-constructed) so you shouldn't be suffering that cost at all. RVO will only make that final value returning a bit cheaper. See for instance https://godbolt.org/z/EvxY6r (gcc 6.2, C++14), where the construction of the pair eventually leads to calling the move constructor of `vector` (see lines 29 and 38 of the generated assembly). And in line 184 it's clear that RVO is even enabled by default, as there's no local `pair` object created.

Reply:
I'll try taking it out again